File: src/Plotly.NET/Traces/ObjectAbstractions/FunnelConnector.fs
namespace Plotly.NET.TraceObjects

open Plotly.NET
open Plotly.NET.LayoutObjects
open DynamicObj
open System
open System.Runtime.InteropServices

/// Styles for connector lines in Funnel Charts.
///
/// Parameters:
///
/// Line          : Sets the Line style for this WaterfallConnector
///
/// Visible       : Whether or not connectors are visible
///
/// ConnectorMode : Sets the shape of connector lines.
type FunnelConnector() =
    inherit DynamicObj()

    static member init
        (
            ?FillColor: Color,
            ?Line: Line,
            ?Visible: bool
        ) =

        FunnelConnector() |> FunnelConnector.style (?FillColor = FillColor, ?Line = Line, ?Visible = Visible)

    static member style
        (
            ?FillColor: Color,
            ?Line: Line,
            ?Visible: bool
        ) =
        fun (connector: FunnelConnector) ->

            connector
            |> DynObj.withOptionalProperty "fillcolor" FillColor
            |> DynObj.withOptionalProperty "line" Line
            |> DynObj.withOptionalProperty "visible" Visible

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/FunnelConnector.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.FunnelConnector FunnelConnector
        (

            Optional<Color> FillColor = default,
            Optional<Line> Line = default,
            Optional<bool> Visible = default
        ) => 
            Plotly.NET.TraceObjects.FunnelConnector.init(

                FillColor: FillColor.ToOption(),
                Line: Line.ToOption(),
                Visible: Visible.ToOption()
            );
}

